src/sentry/lang/javascript/processor.py

```diff
@@ -970,21 +1163,39 @@ def process_frame(self, processable_frame, processing_task):
         ):
             return
 
+        # We get the debug_id for file in this frame, if any.
+        debug_id = self.abs_path_debug_id.get(frame["abs_path"])
+
         # The 'sourceview' is used for pre/post and 'context_line' frame expansion.
         # Here it's pointing to the minified source, however the variable can be shadowed with the original sourceview
-        # (or 'None' if the token doesnt provide us with the 'context_line') down the road.
-        # TODO: add support for debug ids.
-        sourceview = self.get_or_fetch_sourceview(url=frame["abs_path"])
-        sourcemap_cache = self.get_or_fetch_sourcemap_cache(url=frame["abs_path"])
+        # (or 'None' if the token doesn't provide us with the 'context_line') down the road.
+        sourceview, resolved_sv_with_debug_id = self.get_or_fetch_sourceview(
+            url=frame["abs_path"],
+            debug_id=debug_id,
+            source_file_type=SourceFileType.MINIFIED_SOURCE,
+        )
+        # Only if we resolved the minified source with a debug_id we want to try and resolve the sourcemap with the
```

Comment:
NOTE:
We should write down these assumptions somewhere, like:
We only want to use the `debug_id` for sourcemap lookup if the minified file was found via the `debug_id`, etc…

Reply:
Where would you write them down? A common internal document?

Reply:
I would have loved this document too, indeed most assumptions have been either inferred or decided after some reasonings.

@mitsuhiko should we sit down to write down our assumptions?

Reply:
This does not really have anything to do with source bundles. It is rather how sourcemap symbolication is supposed to work, and what kind of data it will try to use under which circumstances. source / artifact bundles are just a tiny part of that.

Reply:
I haven't personally touched most of the core logic of the symbolication so I could help to write the assumptions for the debug_id and url logic.